src/output/plugins/OSXOutputPlugin.cxx

```diff
@@ -841,22 +872,19 @@ OSXOutput::Open(AudioFormat &audio_format)
 					 errormsg);
 	}
 
-	UInt32 buffer_frame_size = 1;
-	status = osx_output_set_buffer_size(au, asbd, &buffer_frame_size);
+	status = osx_output_set_buffer_size(au);
 	if (status != noErr) {
 		osx_os_status_to_cstring(status, errormsg, sizeof(errormsg));
 		throw FormatRuntimeError("Unable to set frame size: %s",
 					 errormsg);
 	}
 
-	size_t ring_buffer_size = std::max<size_t>(buffer_frame_size,
-						   MPD_OSX_BUFFER_TIME_MS * audio_format.GetFrameSize() * audio_format.sample_rate / 1000);
+	size_t ring_buffer_size = MPD_OSX_BUFFER_TIME_MS * audio_format.GetFrameSize() * audio_format.sample_rate / 1000;
 
 #ifdef ENABLE_DSD
-        if (dop_enabled) {
+	if (dop_enabled) {
 		pcm_export->Open(audio_format.format, audio_format.channels, params);
-		ring_buffer_size = std::max<size_t>(buffer_frame_size,
-						   MPD_OSX_BUFFER_TIME_MS * pcm_export->GetOutputFrameSize() * asbd.mSampleRate / 1000);
+		ring_buffer_size = MPD_OSX_BUFFER_TIME_MS * pcm_export->GetOutputFrameSize() * asbd.mSampleRate / 1000;
```

Comment:
Food (for a new commit, not to be mixed into this one): this formula is redundant; use `AudioFormat::TimeToSize()`.
Also I'd prefer to use `std::chrono` instead of those `_MS` suffixes and hard-coding `* 1000`.

Reply:
Will do in a new PR.